### PR TITLE
fix integration json loading parser so that it will trim unneeded emp…

### PR DESCRIPTION
**Pull request by YANG-DB**
trim empty spaces from the ndjson file and match the existing behavior of the savedObject import dashboard that works even when empty spaces appear in the ndjson file

### Description
Integration loading of a dashboard fails when empty spaces appear in the ndjson file

This works well in the savedObject import api

### What is the expected behavior?
Success loading the ndjson dashboard even if it has empty lines - fix the integration parser to trim the ndjson

### Issues Resolved
https://github.com/opensearch-project/dashboards-observability/issues/757

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/adaptors/integrations/repository/integration.ts [hunk 1]
```diff
@@ -211,7 +211,7 @@ export class Integration {
       );
       try {
         const ndjson = await fs.readFile(sobjPath, { encoding: 'utf-8' });
-        const asJson = '[' + ndjson.replace(/\n/g, ',') + ']';
+        const asJson = '[' + ndjson.trim().replace(/\n/g, ',') + ']';
```
Comment (Swiddis) [inline server/adaptors/integrations/repository/integration.ts:214]: This breaks NDJson compliance. The [spec](https://github.com/ndjson/ndjson-spec#32-parsing) says that the two permissible behaviors are to ignore all blank lines, or raise an exception on any blank line. If we want to switch to the former mode of operation, we need to handle all blank lines, not just ones at the start or end of the file.

Comment (Swiddis) [inline server/adaptors/integrations/repository/integration.ts:214]: Should probably include a test case that has an empty line to avoid regressions if we refactor this later.

**Review by Swiddis — CHANGES_REQUESTED** [review]

File: server/adaptors/integrations/repository/integration.ts (hunk 1 above)
Comment (YANG-DB) [reply server/adaptors/integrations/repository/integration.ts:214]: as stated - regardless of the spec it should work identically to how the savedObject import API work - if that works with the empty lines so does this code

Comment (Swiddis) [reply server/adaptors/integrations/repository/integration.ts:214]: Where is the saved object import format specified? It's mentioned in #743 that other implementation quirks make this nontrivial. We shouldn't assume they share the same behavior based on one example.

Comment (YANG-DB) [reply server/adaptors/integrations/repository/integration.ts:214]: why should we test this case ? its not transitioning the data but only removed none necessary empty lines - if none exist it does nothing...
how would we possibly break this in the future ?

**Review by Swiddis — APPROVED** [review]

File: server/adaptors/integrations/repository/integration.ts (hunk 1 above)
Comment (pjfitzgibbons) [reply server/adaptors/integrations/repository/integration.ts:214]: If it's simple to test, "how would we possibly break this in the future" is exactly the kind of unknown to test-proof.  since this is a data-modifying function, I'd lean toward producing some sort of test.

Comment (pjfitzgibbons) [inline server/adaptors/integrations/repository/integration.ts:214]: As commented in resovled : If it's simple to test, "how would we possibly break this in the future" is exactly the kind of unknown to test-proof. since this is a data-modifying function, I'd lean toward producing some sort of test.

**Review by pjfitzgibbons — APPROVED** [review]
